hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java

```diff
@@ -113,6 +111,8 @@
   private int maxRetriesMultiplier;
   // Indicates if this particular source is running
   volatile boolean sourceRunning = false;
+  // Indicates if the source initialization is in progress
+  private volatile boolean startupOngoing = false;
```

Comment:
This is not accessed anywhere?

Should be backport HBASE-24877 and then apply this change?

Reply:
@bharathv yes I see that this was not backported to branch2.3, do you mean creating a new Jira for backporting or can you commit it without jira? I can then merge the latest changes.

Reply:
Ya, let me try backporting it and then you can rebase. will update you once it is committed.